plugins/Wallet/js/addresses.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+// Library for working with clipboard
+const Clipboard = require('clipboard');
+// Keeps track of number of addresses
+var addressCount = 0;
+// Keeps track of if the user is typing into the search bar
+var typing;
+
+// Get transactions for a specific wallet address
+function updateAddrTxn(event) {
+	$('#transaction-list').empty();
+	IPC.sendToHost('api-call', {
+		url: '/wallet/transactions/' + event.target.innerText,
+		type: 'GET',
+	}, 'update-history');
+}
+
+// Make wallet address html element
+function makeAddress(address, callback) {
+	// Create only new addresses
+	if (typeof(address) === 'undefined') { return; }
+	if ($('#' + address).length !== 0) { return; }
+	addressCount++;
+
+	// Make and store a jquery element for the address
+	var addr = $(`
+		<div class='entry' id='` + address + `'>
+			<div class='listnum'>` + addressCount + `</div>
+			<div class='address'>` + address + `</div>
+			<div class='copy-address'><i class='fa fa-clipboard'></i></div>
+		</div>
+	`);
+
+	process.nextTick(function() {
+		callback(addr);
+	});
+}
+
+// Append the wallet address
+function appendAddress(element) {
+	// Make clicking this address show relevant transactions
+	element.find('.address').click(updateAddrTxn);
+	// Make copy-to-clipboard button clickable
+	element.find('.copy-address').click(function() {
+		Clipboard.writeText(this.parentNode.id);
+		notify('Copied address to clipboard', 'copied');
+	});
+
+	// Append and show the address element
+	$('#address-list').append(element);
+}
+
+// Make and append a single address
+function addAddress(address) {
+	makeAddress(address, appendAddress);
+}
+
+// From an array of addresses, make elements then insert into the page in a
+// semi-non-blocking manner
+// TODO: Could this be done better?
```

Comment:
It passes the address to makeAddress() to make the html element, sends the result to appendAddress() to add that html element to the list.

This will likely be redone to paginate viewing of addresses and transactions.